runelite-client/src/main/java/net/runelite/client/plugins/vorkath/VorkathPlugin.java

```diff
@@ -0,0 +1,106 @@
+package net.runelite.client.plugins.vorkath;
+
+import java.time.temporal.ChronoUnit;
+import javax.inject.Inject;
+
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.NPC;
+import net.runelite.api.Query;
+import net.runelite.api.queries.NPCQuery;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.task.Schedule;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.util.QueryRunner;
+
+@PluginDescriptor(
+		name = "Vorkath"
+)
+public class VorkathPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private QueryRunner queryRunner;
+
+	@Inject
+	private VorkathOverlay overlay;
+
+	private VorkathAttack attack;
+
+	int vorkathAttackCounter = 6;
+	String vorkathNextSpecial = "Unknown";
+	private boolean vorkathHasAttacked = false;
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	@Schedule(
+		period = 1000,
+		unit = ChronoUnit.MILLIS
+	)
+
+	public void update()
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		NPC vorkath = findVorkath();
```

Comment:
Maybe it's worth checking whether the player is in the Vorkath instance before checking all NPC's?